### DISCUSSION TITLE
wire:model with @change bug

### DISCUSSION THREAD

**Discussion by sueboy19** [livewire/livewire#4240] [1. Help] [answered]
Web start "B Button" be show.
When select be change, @change action be trigger, "A Button" need be show and "B Button" need hidden.
But result is A & B Button all be show.

```blade
<div class="creatSection" x-data="fun_init_var()">
    <select @change="edit_cheaged()" wire:model="proudction_type_id">
	<option value="1">1</option>
	<option value="2">2</option>
    </select>
</div>

<button x-show="editChanged" type="button">A edit changed</button>
<button x-show="!editChanged" type="button">B edit no changed</button>
				
<script>
fun_init_var() {
	return {
               'editChanged' : false,
		edit_cheaged(){
			this.editChanged = true;   // 多筆編輯用
		},
	},
}
</script>
```

Only select use  wire:model.defer is work fine.

I think is wire:model keep something, then editChanged be restore;

**Comment by joshhanley** [comment, marked as answer]
@sueboy19 there are a couple of issues:

One is you need to make sure your buttons are inside the div with `x-data` on it, otherwise they won't work (see below for changes).

Also you are missing the word `function` from the front of your function definition (again see below).

Hope this helps!

```blade
<div class="creatSection" x-data="fun_init_var()">
    <select @change="edit_cheaged()" wire:model="proudction_type_id">
	<option value="1">1</option>
	<option value="2">2</option>
    </select>

    <button x-show="editChanged" type="button">A edit changed</button>
    <button x-show="!editChanged" type="button">B edit no changed</button>
</div>
				
<script>
function fun_init_var() {
	return {
        'editChanged' : false,
		edit_cheaged(){
			this.editChanged = true;   // 多筆編輯用
		},
	},
}
</script>
```

**Reply by sueboy19** [reply]
>> One is you need to make sure your buttons are inside the div with x-data on it, otherwise they won't work (see below for changes).

You are right. I try and success, just move into div.
